quickwit/quickwit-actors/src/registry.rs

```diff
@@ -64,6 +69,17 @@ impl<A: Actor> JsonObservable for TypedJsonObservable<A> {
         let state: <A as Actor>::ObservableState = oneshot_rx.await.ok()?;
         serde_json::to_value(&state).ok()
     }
+
+    async fn quit(&self) -> Option<ActorExitStatus> {
+        let _ = self.weak_mailbox.upgrade().map(|mailbox| {
+            let _ = mailbox.send_message_with_high_priority(Command::Quit);
+        });
```

Comment:
We tend to avoid using map with side-effects.

```suggestion
        if let Some(mailbox) = self.weak_mailbox.upgrade() {
 let _ = mailbox.send_message_with_high_priority(Command::Quit);
        }
```